### PR TITLE
Autogeneration of report type in service

### PR DESCRIPTION
**Pull request by Anton-Am**
Now developers can create reports without need for type creating. 

For example:

```
_reportsCoordinator.Add(
                        "Email",
                        ReportEntryType.Information,
                        String.Format("User {0} send next message:<br/> <b>Subject:</b> {1} <br/> <b>Message:</b> {2} <br/> <b>Recipients:</b> {3} ", userName, parameters["Subject"], parameters["Body"], parameters["Recipients"]),
                        "Email logs"
                    );
```

### CODE REVIEW

File: src/Orchard.Web/Core/Reports/Views/Admin/Display.cshtml [hunk 1]
```diff
@@ -3,37 +3,34 @@
...
+        </thead>
+        @foreach (var reportEntry in Model.Report.Entries)
+        {
             <tr>
                 <td>
                     @reportEntry.Type
                 </td>
                 <td>
-                    @reportEntry.Message
+                    @Html.Raw(reportEntry.Message)
```
Comment (sebastienros) [inline src/Orchard.Web/Core/Reports/Views/Admin/Display.cshtml]: Red flag

**Comment by sebastienros** [comment]
I think we are deprecating this module, you should use the Log instead.

File: src/Orchard/Reports/Report.cs [hunk 2]
```diff
@@ -7,7 +7,7 @@ public class Report {
             Entries = new List<ReportEntry>();
         }
 
-        public IList<ReportEntry> Entries { get; set;}
+        public List<ReportEntry> Entries { get; set;}
```
Comment (Piedone) [inline src/Orchard/Reports/Report.cs]: Why?

**Comment by sebastienros** [comment]
Please open an issue and start a discussion if you still think it's necessary.

File: src/Orchard.Web/Core/Reports/Views/Admin/Display.cshtml (hunk 1 above)
Comment (Anton-Am) [inline src/Orchard.Web/Core/Reports/Views/Admin/Display.cshtml]: This will help to display formatted messages in report page.

File: src/Orchard/Reports/Report.cs (hunk 2 above)
Comment (Anton-Am) [inline src/Orchard/Reports/Report.cs]: So we avoid the error when adding a new report

Comment (sfmskywalker) [inline src/Orchard/Reports/Report.cs]: Ok but why the change from IList to List?

File: src/Orchard.Web/Core/Reports/Views/Admin/Display.cshtml [hunk 3]
```diff
@@ -3,37 +3,34 @@
...
+        </colgroup>
+        <thead>
+            <tr>
+                <th scope="col">@T("Type")</th>
+                <th scope="col">@T("Message")</th>
+                <th scope="col">@T("Date")</th>
+            </tr>
+        </thead>
+        @foreach (var reportEntry in Model.Report.Entries)
+        {
```
Comment (sfmskywalker) [inline src/Orchard.Web/Core/Reports/Views/Admin/Display.cshtml]: Please format your code so that curly braces start at the end of the line instead of on a new line.

File: src/Orchard/Reports/Report.cs (hunk 2 above)
Comment (Anton-Am) [inline src/Orchard/Reports/Report.cs]: I got exception about fixed size when adding new report

File: src/Orchard.Web/Core/Reports/Views/Admin/Display.cshtml (hunk 3 above)
Comment (Anton-Am) [inline src/Orchard.Web/Core/Reports/Views/Admin/Display.cshtml]: Ok :)

File: src/Orchard/Reports/Report.cs (hunk 2 above)
Comment (sfmskywalker) [inline src/Orchard/Reports/Report.cs]: That must have been something else, e.g. assigning an Array instead of a List. Regardless, the return type should be IList (but initialized to an instance of List as you did). As to why using IList vs List as the type, here's an interesting SO post: http://stackoverflow.com/questions/8717582/why-use-ilist-or-list
